server/src/main/java/org/opensearch/search/aggregations/bucket/composite/CompositeAggregator.java

```diff
@@ -207,6 +234,41 @@ public InternalAggregation[] buildAggregations(long[] owningBucketOrds) throws I
                 aggs
             );
         }
+
+        // Build results from fast filters optimization
+        if (bucketOrds != null) {
+            final Map<CompositeKey, InternalComposite.InternalBucket> bucketMap = new HashMap<>();
+            for (InternalComposite.InternalBucket internalBucket : buckets) {
+                bucketMap.put(internalBucket.getRawKey(), internalBucket);
+            }
+
+            LongKeyedBucketOrds.BucketOrdsEnum ordsEnum = bucketOrds.ordsEnum(0);
+            while (ordsEnum.next()) {
+                Long bucketValue = ordsEnum.value();
+                CompositeKey key = new CompositeKey(bucketValue);
+                if (bucketMap.containsKey(key)) {
+                    long docCount = bucketDocCount(ordsEnum.ord()) + bucketMap.get(key).getDocCount();
+                    bucketMap.get(key).setDocCount(docCount);
+                } else {
```

Comment:
Can you add some comments to explain the logic here?

Reply:
Add more comments